app/uk/gov/hmrc/apisubscriptionfields/repository/FieldsDefinitionRepository.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apisubscriptionfields.repository
+
+import javax.inject.{Inject, Singleton}
+
+import com.google.inject.ImplementedBy
+import play.api.Logger
+import play.api.libs.json._
+import reactivemongo.api.commands.WriteResult
+import reactivemongo.api.indexes.Index
+import reactivemongo.api.indexes.IndexType.Ascending
+import reactivemongo.bson.{BSONDocument, BSONObjectID}
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+//TODO: think about getting rid of trait
+@ImplementedBy(classOf[FieldsDefinitionMongoRepository])
+trait FieldsDefinitionRepository {
+
+  //TODO consider Future[Boolean]
+  def save(fieldsDefinition: FieldsDefinition): Future[Unit]
```

Comment:
`Future[Boolean]` or `Future[Option[FieldsDefinition]]`
Check how much code you need to write, and if we use it.

Reply:
`Future[Boolean]` will be implemented in PR5 so you will be able to review in the next PR.